### ISSUE TITLE
ascontigousarray/asfortranarray differ from numpy

### ISSUE THREAD

**Issue by peterbell10** [cupy/cupy#3511]
In NumPy, `np.ascontiguousarray(x)` and `np.asfortranarray(x)` are equivalent to `np.asarray(x, order={'C', 'F'})`. Meaning it will accept lists and other array-like types to convert them into an array.

In CuPy, the array-like argument must be exactly an array and no coercion is permitted
```python
In [1]: import cupy

In [2]: cupy.asarray(range(10), order='C')                                                   
Out[2]: array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])

In [3]: cupy.ascontiguousarray(range(10))                                                    
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-073bb62b4261> in <module>
----> 1 cupy.ascontiguousarray(range(10))

~/miniconda3/envs/uarray-dev/lib/python3.7/site-packages/cupy/creation/from_data.py in ascontiguousarray(a, dtype)
     95 
     96     """
---> 97     return core.ascontiguousarray(a, dtype)
     98 
     99 

TypeError: Argument 'a' has incorrect type (expected cupy.core.core.ndarray, got range)
```

**Comment by leofang** [comment]
There are many NumPy APIs accepting lists/array-like objects as input, but [CuPy disallows them](https://docs-cupy.chainer.org/en/stable/reference/difference.html#universal-functions-only-work-with-cupy-array-or-scalar). I could be wrong, but my guess is `ascontigousarray()`/`asfortranarray()` are not considered as one of the proper entry points in CuPy.

**Comment by grlee77** [comment]
Yeah, I think the question is whether `cp.ascontiguousarray` and `cp.asfortranarray` should be considered array creation functions like `cp.asarray` and allow this use case.

**Comment by leofang** [comment]
I noticed that NumPy considers `ascontiguousarray()` a creation routine, but not `asfortranarray()`. Interesting...
https://numpy.org/doc/1.18/reference/routines.array-creation.html#from-existing-data

**Comment by kmaehashi** [comment]
This behavior is intended as documented below:

https://docs-cupy.chainer.org/en/latest/reference/generated/cupy.ascontiguousarray.html#cupy.ascontiguousarray

    a (cupy.ndarray) – Source array.

However I think it's better to add some notes in our documentation. cupy.ascontiguousarray and cupy.copy are under `Creation from other data` but only accepts cupy.ndarrays.
https://docs-cupy.chainer.org/en/latest/reference/creation.html#creation-from-other-data
